services/gateway/src/ops_batcher.rs

```diff
@@ -262,18 +283,156 @@ impl OpsBatcherRunner {
+                                                        for c in calls {
+                                                            collect_calls(c, target_hex, out);
+                                                        }
+                                                    }
+                                                }
+                                            }
+
+                                            let target_hex = format!("0x{:x}", registry);
+                                            let mut subcalls: Vec<&serde_json::Value> = Vec::new();
+                                            let root = trace.get("result").unwrap_or(&trace);
+                                            collect_calls(root, &target_hex, &mut subcalls);
+
+                                            // If counts mismatch, fall back to finalize-all to avoid mislabeling.
+                                            if subcalls.len() != ids_for_receipt.len() {
+                                                tracker
+                                                    .set_status_batch(
+                                                        &ids_for_receipt,
+                                                        RequestState::Finalized {
+                                                            tx_hash: hash.clone(),
+                                                        },
+                                                    )
+                                                    .await;
+                                            } else {
+                                                // Map each subcall to success/failure and update corresponding request id.
+                                                for (i, id) in ids_for_receipt.iter().enumerate() {
+                                                    let sc = subcalls[i];
+                                                    let error_str = sc
+                                                        .get("error")
+                                                        .and_then(|e| e.as_str())
+                                                        .map(|s| s.to_string());
+                                                    let failed_flag = sc
+                                                        .get("reverted")
+                                                        .and_then(|b| b.as_bool())
+                                                        .unwrap_or(false)
+                                                        || sc
+                                                            .get("failed")
+                                                            .and_then(|b| b.as_bool())
+                                                            .unwrap_or(false);
+                                                    // Try to extract revert data for better error code parsing.
+                                                    let revert_data = sc
```

Comment:
GPT says that output is always `0x` when it reverts so should always check `revertReason`